presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataReader.java

```diff
@@ -98,12 +98,28 @@ public DwrfMetadataReader(RuntimeStats runtimeStats, OrcReaderOptions readerOpti
     }
 
     @Override
-    public PostScript readPostScript(byte[] data, int offset, int length)
+    public PostScript readPostScript(OrcDataSourceId orcDataSourceId, byte[] data, int offset, int length)
             throws IOException
     {
+        // corrupted or non-ORC files often have some garbage at the end of the file instead of the PostScript length
+        if (length <= 0) {
+            throw new OrcCorruptionException(orcDataSourceId, "Invalid PostScript length %s, likely a corrupted or non-ORC file", length);
+        }
+
         long cpuStart = THREAD_MX_BEAN.getCurrentThreadCpuTime();
         CodedInputStream input = CodedInputStream.newInstance(data, offset, length);
         DwrfProto.PostScript postScript = DwrfProto.PostScript.parseFrom(input);
+        if (!postScript.hasFooterLength()) {
+            throw new OrcCorruptionException(orcDataSourceId, "PostScript doesn't have Footer length, likely a corrupted or non-ORC file");
+        }
+
+        if (postScript.getFooterLength() <= 0) {
+            throw new OrcCorruptionException(orcDataSourceId, "Invalid Footer length %s, likely a corrupted or non-ORC file", postScript.getFooterLength());
+        }
+
+        if (!postScript.hasCompression()) {
```

Comment:
Compression is optional, so is this check right ?

Reply:
Well, this one is tricky. CompressionKind will default to NONE in the PostScript.getCompression if it is not provided.

It will be safer to remove compression related checks to make sure files w/o compression set in the PS are not impacted.

Reply:
Once this is removed and all the test passes, let me know I will merge it in.